File: blog_springBoot/src/main/java/com/synergism/blog/core/favorite/entity/MyFavorite.java
package com.synergism.blog.core.favorite.entity;

import com.synergism.blog.core.collection.entity.Collection;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Setter
@Getter
public class MyFavorite extends Favorite{
    private List<Collection> collectionList;
}

File: blog_vue/src/model/favorite/MyFavorite.ts
import Favorite from "./Favorite";

export default class MyFavorite extends Favorite {
  collectionList: Array<Collection>;

  constructor() {
    super();
    this.collectionList = [new Collection()];
  }
}

class Collection {
  id: number;
  title: string;
  href: string;
  synopsis: string;

  constructor() {
    this.id = -1;
    this.title = "";
    this.href = "";
    this.synopsis = "";
  }
}
